pkg/relayer/proxy/proxy.go

```diff
@@ -75,30 +76,40 @@ type relayerProxy struct {
 	supplierAddress string
 }
 
+// NewRelayerProxy creates a new relayer proxy with the given dependencies or returns
+// an error if the dependencies fail to resolve or the options are invalid.
 func NewRelayerProxy(
-	clientCtx sdkclient.Context,
-	keyName string,
-	keyring keyring.Keyring,
-	proxiedServicesEndpoints servicesEndpointsMap,
-	blockClient blocktypes.BlockClient,
-) relayer.RelayerProxy {
-	accountQuerier := accounttypes.NewQueryClient(clientCtx)
-	supplierQuerier := suppliertypes.NewQueryClient(clientCtx)
-	sessionQuerier := sessiontypes.NewQueryClient(clientCtx)
+	deps depinject.Config,
+	opts ...relayer.RelayerProxyOption,
+) (relayer.RelayerProxy, error) {
+	rp := &relayerProxy{}
+
+	if err := depinject.Inject(
+		deps,
+		&rp.clientCtx,
+		&rp.blockClient,
+	); err != nil {
+		return nil, err
+	}
+
 	servedRelays, servedRelaysProducer := channel.NewObservable[*types.Relay]()
 
-	return &relayerProxy{
-		blockClient:              blockClient,
-		keyName:                  keyName,
-		keyring:                  keyring,
-		accountsQuerier:          accountQuerier,
-		supplierQuerier:          supplierQuerier,
-		sessionQuerier:           sessionQuerier,
-		proxiedServicesEndpoints: proxiedServicesEndpoints,
-		servedRelays:             servedRelays,
-		servedRelaysProducer:     servedRelaysProducer,
-		clientCtx:                clientCtx,
+	rp.servedRelays = servedRelays
+	rp.servedRelaysProducer = servedRelaysProducer
+	rp.accountsQuerier = accounttypes.NewQueryClient(rp.clientCtx)
+	rp.supplierQuerier = suppliertypes.NewQueryClient(rp.clientCtx)
+	rp.sessionQuerier = sessiontypes.NewQueryClient(rp.clientCtx)
+	rp.keyring = rp.clientCtx.Keyring
```

Comment:
Nice! I didn't realize that `client.Context#Keyring` was a thing; TIL. :raised_hands: